test/gtest/ucp/test_ucp_tag.cc

```diff
@@ -423,22 +424,116 @@ class test_ucp_tag_limits : public test_ucp_tag {
 public:
     test_ucp_tag_limits() {
         m_test_offload = get_variant_value();
-        m_env.push_back(new ucs::scoped_setenv("UCX_RC_TM_ENABLE",
-                                               ucs::to_string(m_test_offload).c_str()));
+        if (m_test_offload) {
+            m_env.push_back(new ucs::scoped_setenv("UCX_RC_TM_ENABLE", "y"));
+        }
     }
 
     void init() {
+        /* TODO: Currently all the tests are for intra-node communication only.
+         * Find a way to create inter-node endpoint on a single node */
         test_ucp_tag::init();
-        check_offload_support(m_test_offload);
+
+        if (m_test_offload) {
+            check_offload_support(m_test_offload);
+        }
```

Comment:
Seems the check is not needed. 

```suggestion
        check_offload_support(m_test_offload);
```